seisbench/models/base.py

```diff
@@ -192,6 +193,113 @@ def download_metadata_callback(metadata_path):
 
         return weights
 
+    @classmethod
+    def load(cls, path):
+        """
+        Load a SeisBench model from local path.
+
+        For more information on the SeisBench model format see py:func:`save`.
+
+        :param path: Define the path to the SeisBench model directory.
+        :type path: pathlib.Path
+        :return: Model instance
+        :rtype: SeisBenchModel
+        """
+        model_files = os.listdir(path)
+
+        if sum([file.endswith("pt") for file in model_files]) > 1:
+            seisbench.logger.warning(
+                f"Strange format of SeisBench model directory detected at {path}. "
+                f"There is more than one .pt file. This will lead to unforseen "
+                f"behaviour during model loading."
+            )
+        if sum([file.endswith("json") for file in model_files]) > 1:
+            seisbench.logger.warning(
+                f"Strange format of SeisBench model directory detected at {path}. "
+                f"There is more than one .json file. This will lead to unforseen "
+                f"behaviour during model loading."
+            )
+
+        for file in model_files:
+            if file.endswith("json"):
+                with open(path / file, "r") as json_file:
+                    weights_metadata = json.load(json_file)
+            elif file.endswith("pt"):
+                model_weights = torch.load(f"{path / file}")
+
+        def _get_args(obj):
+            signature = inspect.signature(obj)
+            args = {k: v._default for k, v in signature.parameters.items()}
+            if "kwargs" in args.keys():
+                del args["kwargs"]
+            return args
+
+        model_args = weights_metadata.get("model_args", {})
+
+        # Overwrite default class args from config model args
+        cls_args = _get_args(cls)
+        cls_args.update({k: model_args[k] for k in cls_args if k in model_args})
+
+        model = cls(**cls_args)
+
+        model._weights_metadata = weights_metadata
+        model._parse_metadata()
+
+        model.load_state_dict(model_weights)
+
+        return model
+
+    def save(self, path=None, name=None):
+        """
+        Save a SeisBench model locally.
+
+        SeisBench models are stored inside the directory 'path'. SeisBench models are saved in 2 parts,
```

Comment:
So far, we do not identify models by their path but by their name. For example, the pretrained models for GPD are all in the same path, but called `stead.json/stead.pt`, `ethz.json/ethz.pt`, ... . I'd suggest to keep it that way. This differs from datasets, where we assumed one directory per dataset, but I think this is reasonable because model weights are a lot smaller in size.

For saving/loading this would mean that a user just provides one path (to either json, pt, or simply without extension) and the files are than loaded automatically.

Reply:
Thanks for the comment. This makes sense and it would then just be a case of aligning both load / save functions to use ```path``` param. I can edit the load fn to do this.